libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -144,98 +160,152 @@ void elasticsearch_plugin_impl::update_account_histories( const signed_block& b
-   uint32_t limit_documents = 0;
-   if((fc::time_point::now() - b.timestamp) < fc::seconds(30))
-      limit_documents = _elasticsearch_bulk_sync;
-   else
-      limit_documents = _elasticsearch_bulk_replay;
+void elasticsearch_plugin_impl::doVisitor(const optional <operation_history_object>& oho)
+{
+   operation_visitor o_v;
+   oho->op.visit(o_v);
+
+   vs.fee_data.asset = o_v.fee_asset;
+   vs.fee_data.amount = o_v.fee_amount;
+
+   vs.transfer_data.asset = o_v.transfer_asset_id;
+   vs.transfer_data.amount = o_v.transfer_amount;
+   vs.transfer_data.from = o_v.transfer_from;
+   vs.transfer_data.to = o_v.transfer_to;
+
+   vs.fill_data.order_id = o_v.fill_order_id;
+   vs.fill_data.account_id = o_v.fill_account_id;
+   vs.fill_data.pays_asset_id = o_v.fill_pays_asset_id;
+   vs.fill_data.pays_amount = o_v.fill_pays_amount;
+   vs.fill_data.receives_asset_id = o_v.fill_receives_asset_id;
+   vs.fill_data.receives_amount = o_v.fill_receives_amount;
+   vs.fill_data.fill_price = o_v.fill_fill_price;
+   vs.fill_data.is_maker = o_v.fill_is_maker;
+}
+
+bool elasticsearch_plugin_impl::add_elasticsearch( const account_id_type account_id,
+                                                   const optional <operation_history_object>& oho,
+                                                   const signed_block& b)
+{
+   const auto &stats_obj = getStatsObject(account_id);
+   const auto &ath = addNewEntry(stats_obj, account_id, oho);
+   growStats(stats_obj, ath);
+   createBulkLine(ath);
+   prepareBulk(ath.id);
+   cleanObjects(ath, account_id);
+
+   if (curl && bulk_lines.size() >= limit_documents) { // we are in bulk time, ready to add data to elasticsearech
+      prepare.clear();
 
-   // we have everything, creating bulk line
-   bulk_struct bulks;
-   bulks.account_history = ath;
-   bulks.operation_history = os;
-   bulks.operation_type = op_type;
-   bulks.block_data = bs;
-   bulks.additional_data = vs;
+      es.curl = curl;
+      es.bulk_lines = bulk_lines;
+      es.elasticsearch_url = _elasticsearch_node_url;
+      es.auth = _elasticsearch_basic_auth;
 
-   std::string data = fc::json::to_string(bulks);
+      if(!graphene::utilities::SendBulk(es))
+         return false;
+      else
+         bulk_lines.clear();
+   }
 
-   auto block_date = bulks.block_data.block_time.to_iso_string();
-   std::vector<std::string> parts;
-   boost::split(parts, block_date, boost::is_any_of("-"));
-   std::string index_name = "graphene-" + parts[0] + "-" + parts[1]; // index name
-   std::string _id = fc::json::to_string(ath.id);
+   return true;
+}
 
-   prepare = graphene::utilities::createBulk(index_name, data, _id, 0);
-   bulk.insert(bulk.end(), prepare.begin(), prepare.end());
+const account_statistics_object& elasticsearch_plugin_impl::getStatsObject(const account_id_type account_id)
+{
+   graphene::chain::database& db = database();
+   const auto &stats_obj = account_id(db).statistics(db);
```

Comment:
Rebase to latest develop branch, use `database::get_account_stats_by_owner()` for better performance.